agents/a2a_mcp_integration.py

```
@@ -427,6 +425,71 @@
+            # Make real HTTP call to MCP server
+            async with aiohttp.ClientSession() as session:
+                async with session.post(
+                    f"{mcp_url}/tools/{tool_name}",
+                    json={
+                        "jsonrpc": "2.0",
+                        "method": "tools/call",
+                        "params": {"name": tool_name, "arguments": params},
+                        "id": 1,
+                    },
+                    timeout=aiohttp.ClientTimeout(total=30),
+                ) as response:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)